joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/astcreation/AstForTypesCreator.scala

```diff
@@ -0,0 +1,163 @@
+package io.joern.php2cpg.astcreation
+
+import io.joern.php2cpg.astcreation.AstCreator.TypeConstants
+import io.joern.php2cpg.parser.Domain.*
+import io.joern.x2cpg.{Ast, ValidationMode}
+import io.shiftleft.codepropertygraph.generated.ModifierTypes
+import io.shiftleft.semanticcpg.language.types.structure.NamespaceTraversal
+
+trait AstForTypesCreator(implicit withSchemaValidation: ValidationMode) { this: AstCreator =>
+
+  protected def astForClassLikeStmt(stmt: PhpClassLikeStmt): Ast = {
+    stmt.name match {
+      case None       => astForAnonymousClass(stmt)
+      case Some(name) => astForNamedClass(stmt, name)
+    }
+  }
+
+  private def astForAnonymousClass(stmt: PhpClassLikeStmt): Ast = {
+    // TODO
+    Ast()
```

Comment:
Could we return a NewUnknown and a debug log message just like the other frontends do for AST elements not yet implemented / supported?

Reply:
This applies to all other similar code locations as well of course